File: draw_figures.rb
require_relative 'taciturn_wookie'

def mean(array)
  array.inject(0) { |sum, x| sum += x } / array.size.to_f
end

rng = Random.new(127705820771904821112999033422058165911)

poisson_parameter_values = [1,4,10,25,50,100,500]
clients_values = [10,100,500,1000, 5000, 10000]
first_law_parameter = 30
second_law_parameter = 0.1

poisson_results = []
poisson_crude_results = []

poisson_parameter_values.each do |param|
  clients_values.each do |clients_number|
    w = TaciturnWookie::Worker.new(clients_number, first_law_parameter, second_law_parameter, rng)
    values = w.proceed do
      rng.poisson(param)
    end
    poisson_crude_results << {
      poisson_parameter: param,
      clients_number: clients_number,
      values: values
    }
    actual_values = values.map {|e| e[2]}
    poisson_results << {
      poisson_parameter: param,
      clients_number: clients_number,
      mean: mean(actual_values)
    }
  end
end

p poisson_results

File: draw_figures.py
from numpy.random import RandomState
import numpy as np
import matplotlib.pyplot as plt
from taciturn_wookie import *


prng = RandomState(131252476)
poisson_parameters = [25, 30]
clients_numbers = [10, 100, 1000, 10000, 100000]

exponential_results = []
exponential_crude_values = []
exponential_parameter = 10
uniform_parameter = 30

for poisson_parameter in poisson_parameters:
    for clients_number in clients_numbers:

        w = TaciturnWorker(second_law_name = prng.exponential,
            first_law_parameter = uniform_parameter,
            second_law_parameter = exponential_parameter, 
            rng = prng, 
            poisson_parameter = poisson_parameter, 
            clients_number = clients_number)
        values = w.proceed()
        response_time_values = values['response_time_values']
                
        mean = reduce(lambda x, y: x + y, response_time_values)/clients_number
        exponential_results.append({
            'second_law': 'exponential',
            'clients_number': clients_number,
            'poisson_parameter': poisson_parameter,
            'mean': mean,
            'response_time_values':response_time_values,
            'clients_number_per_time_values':values['clients_number_per_time_values']
        })
        
plot_from = filter(lambda x: x['clients_number'] == clients_numbers[-1], exponential_results)

for result in plot_from:
    clients_count = map(lambda x: x[0], result['clients_number_per_time_values'])
    clients_time = map(lambda x: x[1], result['clients_number_per_time_values'])
    plt.plot(clients_time, clients_count, label="$\lambda = %d$" % result['poisson_parameter'])
            
plt.legend(shadow=True, fancybox=True)
plt.savefig('plot.png', dpi=96)